### PR TITLE
Enhanced render props

### PR DESCRIPTION
**Pull request by jondashkyle**
I pulled the `render-props` branch into my development environment for Mirror and used it to glue our existing NFT component to `NFTE`. It worked like a charm! Along the way I introduced a few changes:

- Made certain params optional for cleaner initialization
- Export `types.ts` to `dist` for use when `importing`

This is about it. Overall, it was extremely simple!

### CODE REVIEW

File: packages/react/package.json [hunk 1]
```diff
@@ -13,8 +13,8 @@
   "author": "Sam Mason de Caires <sam.jbmason@gmail.com>",
   "repository": "sammdec/nfte",
   "scripts": {
-    "build": "microbundle -f es,cjs --jsx React.createElement --define API_HOST='https://nfte.app' --css inline",
```
Comment (sammdec) [inline packages/react/package.json]: Would you mind swithcing these back around

File: packages/react/src/index.tsx [hunk 2]
```diff
@@ -112,26 +112,55 @@ function NFT({
...
+    className?: string
+    style?: CSSProperties
+    darkMode?: boolean
+    autoPlay?: boolean
+  }) => JSX.Element
 }) {
-  useStyleSheet(styles)
-
   const [data, setData] = useState<NFTData | undefined>(initialData)
+  const element = children || Embed
```
Comment (sammdec) [inline packages/react/src/index.tsx:163]: I went back and forth on this, my worry is that if this logic was here then the code would still be pulled in even if the Embed component was not being used, right?

File: packages/react/tsconfig.json [hunk 3]
```diff
@@ -5,7 +5,11 @@
     "target": "ESNext",
     "esModuleInterop": true,
     "moduleResolution": "node",
+    "declaration": true,
```
Comment (sammdec) [inline packages/react/tsconfig.json:8]: I'm a bit of a ts noob what is `declaration` doing and what does the `include` prop do?

Comment (jondashkyle) [reply packages/react/tsconfig.json:8]: I’m somewhat new too, but this resolved an issue where `/types.ts` was not being moved to `/dist`, preventing it from being possible to `import` the types outside of the component.

I was referencing this issue, and did a copy paste until things worked:
https://github.com/developit/microbundle/issues/58

File: packages/react/src/index.tsx (hunk 2 above)
Comment (jondashkyle) [reply packages/react/src/index.tsx:163]: I think that’s ok considering the size of `Embed`. Needing to import `Embed` increases the scope of use. Ideally it’s possible to initialize the component with just the `address` and `id` in the simplest configuration.

File: packages/react/package.json (hunk 1 above)
Comment (jondashkyle) [reply packages/react/package.json]: Good catch! I swapped this out so I could run our Nextjs build in parallel on port `3000` while still being able to query the API endpoint :)

File: packages/react/src/index.tsx (hunk 2 above)
Comment (sammdec) [reply packages/react/src/index.tsx:163]: Yeah fair point

File: packages/react/tsconfig.json (hunk 3 above)
Comment (sammdec) [reply packages/react/tsconfig.json:8]: Weird I can see the delcaration files in my build wihtout having that option turned on

Comment (jondashkyle) [reply packages/react/tsconfig.json:8]: The declaration files appear, but `types.ts` is not imported for some reason, unless you use `include`. I needed access to `types.ts` in order to define the props for my custom `Embed` component.

**Comment by jondashkyle** [comment]
Closing this in favor of `1.0.0` :)

**Pull request closed without merging**